packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -620,6 +621,238 @@ abstract class DialogWindowController extends BaseWindowController {
   void setMinimized(bool minimized);
 }
 
+/// Delegate class for tooltip window controller.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+///
+/// See also:
+///
+/// * [TooltipWindowController], the controller that creates and manages tooltip windows.
+/// * [TooltipWindow], the widget for a tooltip window.
+/// * [RegularWindowControllerDelegate], the delegate for regular window controllers.
+mixin class TooltipWindowControllerDelegate {
+  /// Invoked when the user attempts to close the window.
+  ///
+  /// The default implementation destroys the window. Subclasses
+  /// can override the behavior to delay or prevent the window from closing.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  ///
+  /// * [onWindowDestroyed], which is invoked after the window is closed.
+  @internal
+  void onWindowCloseRequested(TooltipWindowController controller) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+
+    controller.destroy();
+  }
+
+  /// Invoked after the window is closed.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  ///
+  /// * [onWindowCloseRequested], which is invoked when the user attempts to close the window.
+  @internal
+  void onWindowDestroyed() {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+  }
+}
+
+/// A controller for a tooltip window.
+///
+/// A tooltip window is a small window that displays brief, informative text
+/// when a user hovers over or focuses on a UI element. Tooltip windows are
+/// typically used to provide additional context or explanations for UI elements
+/// without cluttering the main interface. As such, the a may not receive input
+/// focus from the user. It will however stay open when another window receives
+/// input focus.
+///
+/// This class does not interact with the widget tree. Instead, it is typically
+/// provided to the [TooltipWindow] widget, which renders the content inside the
+/// tooltip window.
+///
+/// The user of this class is responsible for managing the lifecycle of the window.
+/// When the window is no longer needed, the user should call [destroy] on this
+/// controller to release the resources associated with the window.
+///
+/// {@tool snippet}
+/// An example usage might look like:
+///
+/// ```dart
```

Comment:
@justinmc This is a large sample, what do you think of moving this to switching to a `@tool dartpad` that points to something like `examples/api/lib/widgets/windows/tooltip.0.dart`?

If we do this, some things we need to figure out:

1. The `api` example app will create two top level windows (one from Dart, one from native entry point). Is this OK?
2. How do we properly call out that this is a sample that shows off experimental APIs? I suspect we'll want some sort of big disclaimer comment at the top.
3. How do we properly explain that this sample is only runnable on the main channel with the windowing feature flag? I suspect we might want to assert this in the `main` method.